include/deal.II/lac/block_matrix_base.h

```diff
@@ -1495,7 +1495,12 @@ template <typename MatrixType>
 inline
 BlockMatrixBase<MatrixType>::~BlockMatrixBase ()
 {
-  clear ();
+  try
+    {
+      clear ();
+    }
+  catch (...)
+    {}
```

Comment:
An `Assert` here is clearly a problem.
On the other hand, `AssertNothrow` does never throw. I implemented this macro exactly because of asserts in destructors. So can you please check whether, catching and issuing `AssertNothrow` works?
```
try { ...} catch (...) {AssertNothrow ...}
```